payjoin/src/receive/multiparty/mod.rs

```diff
@@ -248,3 +248,68 @@ impl FinalizedProposal {
 
     pub fn v2(&self) -> &[v2::UncheckedProposal] { &self.v2_proposals }
 }
+
+#[cfg(test)]
+mod test {
+
+    use std::any::{Any, TypeId};
+
+    use payjoin_test_utils::{BoxError, PARSED_ORIGINAL_PSBT};
+
+    use super::{v1, v2, FinalizedProposal, UncheckedProposalBuilder};
+    use crate::receive::optional_parameters::Params;
+    use crate::receive::v2::test::SHARED_CONTEXT;
+
+    fn multiparty_proposal_from_test_vector() -> v1::UncheckedProposal {
+        let pairs = url::form_urlencoded::parse("v=2&optimisticmerge=true".as_bytes());
+        let params =
+            Params::from_query_pairs(pairs, &[2]).expect("Could not parse from query pairs");
```

Comment:
I stared at this `&[2]` for a while wondering what it was. Makes me wonder if we ought to have a `Version` enum.